src/jvmMain/kotlin/org/koitharu/kotatsu_dl/logic/downloader/DownloadTask.kt

```diff
@@ -132,7 +134,7 @@ class DownloadTask(
 	}
 
 	private fun getFileExtensionFromUrl(url: String): String {
-		return url.substringAfterLast('.', "")
+		return url.substringAfterLast('.', "").take(3)
```

Comment:
```suggestion
		return url.substringBeforeLast('?').substringAfterLast('.', "")
```
File extension can also be 4 characters (jpeg for example)

Reply:
yea, sometimes might be 4. Thanks for correct it :D!